### PR TITLE
Parallel it network leak

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Correctly shutdown docker network after the test finidhed 
Why should this be merged: 
 prevent leaking docker networks
Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: it/src/main/resources/logback.xml [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <logger name="org.corfudb.runtime.view.workflows.WorkflowRequest" level="off"/>
 
     <!-- Control logging levels for individual components here. -->
-    <root level="warn">
+    <root level="info">
         <appender-ref ref="STDOUT"/>
```
[the remark this answers is not in the document]
Comment (xnull) [reply it/src/main/resources/logback.xml]: Done

File: it/src/main/java/org/corfudb/universe/util/DockerManager.java [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.universe.node.NodeException;
+
+import java.time.Duration;
+
+@Builder
+@Slf4j
+public class DockerManager {
```
[the remark this answers is not in the document]
Comment (xnull) [reply it/src/main/java/org/corfudb/universe/util/DockerManager.java]: Done

**Comment by xnull** [comment]
@WenbinZhu and just improved it a little bit more https://github.com/CorfuDB/CorfuDB/commit/b0ad6b4

**Comment by WenbinZhu** [comment]
> @WenbinZhu actually no. These prevent the leaks:
> 
> * [CorfuDB/it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java](https://github.com/CorfuDB/CorfuDB/blob/9e863fd0198c96891c95edc17fae6eab17b64338/it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java#L64)
>       
>       
>            Line 64
>         in
>         [9e863fd](/CorfuDB/CorfuDB/commit/9e863fd0198c96891c95edc17fae6eab17b64338)
>       
>       
>       
>       
>   
>           
>             
>              universe = UNIVERSE_FACTORY.buildVmUniverse(universeParams, manager);
> 
> we create the universe, then deploy, so, the universe is not null, we can shutdown it properly
> 
> * Just added a handler to notify if we have a leak 
>     
>       
>         [CorfuDB/it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java](https://github.com/CorfuDB/CorfuDB/blob/a50f676868d0cd615891c58ac81af343c63a4ec8/it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java#L38)
>       
>       
>            Line 38
>         in
>         [a50f676](/CorfuDB/CorfuDB/commit/a50f676868d0cd615891c58ac81af343c63a4ec8)
>       
>       
>       
>       
>   
>           
>             
>              throw new IllegalStateException("The universe is null, can't shutdown the test properly. " +

Ok. Please add a one-line comment before this line so we won't forget later: https://github.com/CorfuDB/CorfuDB/blob/b0ad6b4847c850236a210b23894fafab943067b6/it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java#L77

**Comment by xnull** [comment]
@WenbinZhu I throw an exception, we won't forget

**Comment by xnull** [comment]
And, yes, I will add a comment, it's a good thing anyway

**Comment by xnull** [comment]
Guys if you are ok please approve the PR or ask for changes

**Comment by xnull** [comment]
Guys please, approve or ask for changes!!!

File: it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java [hunk 3]
```diff
@@ -61,9 +64,9 @@ public Scenario getVmScenario(int numNodes) {
                 .universeParams(universeParams)
                 .build();
 
-        universe = UNIVERSE_FACTORY
-                .buildVmUniverse(universeParams, manager)
-                .deploy();
+        //Assign universe variable before deploy prevents resources leaks
```
Comment (WenbinZhu) [inline it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java]: NIT: add a space after "//" and also copy this comment to getDockerScenario() before `UNIVERSE_FACTORY.buildDockerUniverse`

Comment (xnull) [reply it/src/test/java/org/corfudb/universe/GenericIntegrationTest.java]: done

**Pull request merged**